File: Ex13.java
package Ex13;

class Matematica {

    public static long fatorial(int numero) {
        if (numero < 0) {
            throw new IllegalArgumentException("Número deve ser não negativo.");
        }
        long resultado = 1;
        for (int i = 2; i <= numero; i++) {
            resultado *= i;
        }
        return resultado;
    }

    public static void main(String[] args) {
        int numero = 5;
        long resultadoFatorial = Matematica.fatorial(numero);
        System.out.println("Fatorial de " + numero + " é: " + resultadoFatorial);

        numero = 0;
        resultadoFatorial = Matematica.fatorial(numero);
        System.out.println("Fatorial de " + numero + " é: " + resultadoFatorial);
    }
}

File: ex13.go
package main

import (
	"fmt"
)

type Matematica struct{}

func (m Matematica) calcularFatorial(n int) int {
	if n == 0 {
		return 1
	}
	return n * m.calcularFatorial(n-1)
}

func Fatorial(n int) int {
	if n == 0 {
		return 1
	}
	resultado := 1
	for i := 1; i <= n; i++ {
		resultado *= i
	}
	return resultado
}

func main() {
	matematica := Matematica{}
	numero := 5

	fatorialRecursivo := matematica.calcularFatorial(numero)
	fatorialIterativo := Fatorial(numero)

	fmt.Printf("Fatorial (Recursivo) de %d: %d\n", numero, fatorialRecursivo)
	fmt.Printf("Fatorial (Iterativo) de %d: %d\n", numero, fatorialIterativo)
}
